ntbs-service/wwwroot/source/Components/ValidateMultiple.ts

```diff
@@ -7,38 +7,56 @@ const qs = require("qs");
 const ValidateMultiple = Vue.extend({
     props: {
         model: String,
-        properties: Array
+        properties: Array,
+        isDateValidation: Boolean
     },
     data: function () {
         return {
-            hasError: false
+            hasError: false,
+            errorFields: []
         }
     },
     methods: {
         // Annotate values to compare with ref="input[i]", and error fields with ref="errorField[i]", for any number of inputs that should be validated together
         // Each error message will be mapped to the corresponding field.
         validate: function () {
             const inputs = this.createArrayFromRefElements("input");
-            var inputValues: any;
-            if (inputs[0].type === "radio") {
+            var queryString: any;
+            if (this.isDateValidation) {
+                var dayInputs = inputs.map((i: any) => i.$refs.dayInput.value);

```

Comment:
It's a shame we've got all the `(i: any)`s around. If we give `createArrayFromRefElements` a good type then we shouldn't need them?

Reply:
The problem is that the ref elements will refer to completely different objects depending on where they're used - either a vue component, or a radio tag, or input tag etc (if we reuse in future).. I appreciate it's not very clear like this but it also sounds very fiddly making it work otherwise? I'm not super experienced with typescript so not quite sure what the best way forward it

Reply:
Ah yes, the refs could be a little problematic.. 
We *could* try and supply as much info as possible like so:
```
const vueInputs: Array<Vue> = inputs // In the case of date validation, we expect all inputs to be vue components
                var dayInputs = vueInputs.map(i => (<HTMLInputElement> i.$refs.dayInput).value);
                if (arrayContainsEmptyValues(dayInputs)) {
                    return;
                }
```
It's not helping with verbosity much, but it does help with stating the assumptions about the what we're interacting with a bit-  what do you think?

Reply:
yes that's much better - thanks for pointing me in the right direction! Have fixed and will be in my next review.